tests/unit/test_network_connections.py

```diff
@@ -3670,6 +3707,94 @@ def test_path_to_glib_bytes(self):
         self.assertEqual(result.get_data(), b"file:///my/test/path\x00")
 
 
+class TestValidatorMatch(unittest.TestCase):
+    def setUp(self):
+        self.test_profile = {
+            "name": "test",
+            "type": "ethernet",
+            "ip": {
+                "dhcp4": False,
+                "address": "192.168.245.7/24",
+            },
+            "match": None,
+        }
+        self.validator = network_lsr.argument_validator.ArgValidator_DictConnection()
+
+    def test_match_path_empty_list(self):
+        """
+        I can remove a match path setting by setting it to empty list.
+        """
+        self.test_profile["match"] = {"path": []}
+        result = self.validator.validate(self.test_profile)
+        self.assertEqual(result["match"], {"path": []})
+
+    def test_match_path_empty_string_in_list(self):
+        """
+        Test that when 'match.path' setting is [""], a ValidationError raises.
+        """
+        self.test_profile["match"] = {"path": [""]}
+        self.assertRaises(ValidationError, self.validator.validate, self.test_profile)
```

Comment:
I am failing to see which code is triggering this exception, can you please tell me? Also, IMHO it would be nicer to accept this value, too, and treat it like an empty list to simplify Jinja2 templating, too. So that

```
path: None
path: []
path: [""]
path: [None]
path: ""
```

are all normalized into `path: []` - I believe this is already true for most values. Please add test for all these cases, too. Maybe change `test_match_empty_list()` to

```
for disabled_path in (None, [], [""], [None], ""):
    self.test_profile["match"] = {"path": disabled_path}
    result = self.validator.validate(self.test_profile)
    self.assertEqual(result["match"], {"path": []})
```

A cleaner way seems to be to do this with https://docs.python.org/3/library/unittest.html#unittest.TestCase.subTest but then it will need some additional work to ensure it won't fail with Python 2, so for now it seems to be appropriate to do the iteration inside one test case.

Reply:
`path: ""` should be already normalize, because a plain string is split by whitespaces.

```python
        if isinstance(value, Util.STRING_TYPE):
            # we expect a list. However, for convenience allow to
            # specify a string, separated by space. Escaping is
            # not supported. If you need that, define a proper list.
            value = [s for s in value.split(" ") if s]
```


But `path: [""]` and `path: [None]` are not lists that should be normalized to `[]`.

Reply:
yes, the argument validator for lists does not need to normalize it. From what I understood the validator for path could normalize it.

Reply:
Updated.  `None, [], [""], [None] or ""` will be normalized into `[]` for match.path setting